bots-deployment-detector/src/agent.spec.ts

```diff
@@ -0,0 +1,159 @@
+import { provideHandleTransaction } from './agent';
+
+import { Finding, FindingSeverity, FindingType, HandleTransaction, ethers } from 'forta-agent';
+
+import { createAddress } from 'forta-agent-tools';
+import { TestTransactionEvent } from 'forta-agent-tools/lib/test';
+
+import { eventAgentEnabled, eventAgentUpdated, createAgentABI } from './utils';
+
+describe('Bot Deployment Detector Agent', () => {
+  let handleTransaction: HandleTransaction;
+
+  const mockDeployerAdddress = createAddress('0x4');
+  const mockContractAddress = createAddress('0x5');
+  const mockRandomAddress = createAddress('0x8');
+
+  const mockAgentId = '34567';
+  const mockChainId = 137;
+  const eventTypes = ['AgentCreated', 'AgentUpdated', 'AgentEnabled', 'AgentDisabled'];
+
+  beforeAll(() => {
+    handleTransaction = provideHandleTransaction(mockDeployerAdddress, mockContractAddress);
+  });
+
+  describe('handleTransaction', () => {
+    describe('Ignores Transaction', () => {
```

Comment:
although this is not wrong, our approach is using just one `describe` block and putting all negative and positive cases into it.